netket/hilbert/fock.py

```diff
@@ -165,3 +161,17 @@ def states_to_local_indices(self, x):
     @property
     def _attrs(self):
         return (self.size, self._n_max, self._n_particles)
+
+    @property
+    def is_indexable(self) -> bool:
+        """Whether the space can be indexed with an integer"""
+        if not self.is_finite:
+            return False
+        elif _is_indexable(self.shape):
+            # includes constrained with prod(shape) < n_combs
+            return True
+        elif self.constrained:
+            n_combs = math.comb((np.array(self.shape) - 1).sum(), self.n_particles)
+            return _is_indexable(float(n_combs))
```

Comment:
You are checking if the total number of combinations is less than maxfloat, right?

add a comment plz (and why do you convert to float?)